server/data/data-jmap/src/main/java/org/apache/james/jmap/api/vacation/Vacation.java

```diff
@@ -78,23 +84,26 @@ public Builder copy(Vacation vacation) {
         }
 
         public Vacation build() {
-            Preconditions.checkNotNull(textBody);
-            return new Vacation(isEnabled.orElse(DEFAULT_DISABLED), fromDate, toDate, textBody, subject);
+            Preconditions.checkState(textBody.isPresent() || htmlBody.isPresent() || !isEnabled.orElse(DEFAULT_DISABLED), "textBody or htmlBody property of vacationResponse object should not be null when enabled");
```

Comment:
you should first compute the `isEnable` then have a `if` then check `textBody || htmlBody`